ignite/services/plugin/plugin.go

```diff
@@ -152,11 +154,18 @@ func newPlugin(pluginsDir string, cp pluginsconfig.Plugin, options ...Option) *P
 	}
 	p.repoPath = path.Join(parts[:3]...)
 	p.cloneURL, _ = xurl.HTTPS(p.repoPath)
+
+	// Use the beginning of the hash as plugin directory name suffix to avoid
+	// using branch or tag reference names in the path.
+	checksum := md5.Sum([]byte(p.repoPath + p.reference))
+	hash := hex.EncodeToString(checksum[:])
+	p.cloneDir = path.Join(pluginsDir, fmt.Sprintf("%s-%s", p.repoPath, hash[:10]))
+
 	if len(p.reference) > 0 {
 		p.repoPath += "@" + p.reference
 	}
-	p.cloneDir = path.Join(pluginsDir, p.repoPath)
-	p.srcPath = path.Join(pluginsDir, p.repoPath, path.Join(parts[3:]...))
+
+	p.srcPath = path.Join(p.cloneDir, path.Join(parts[3:]...))
```

Comment:
```suggestion
	p.srcPath = path.Join(p.cloneDir, path.Join(parts[3:]...))
```

Can assign set `path.Join(parts[3:]...)` to a variable?  Just for clarity when reading the code.